kfs-core/src/main/java/edu/arizona/kfs/fp/batch/service/impl/ProcurementCardHolderUpdateServiceImpl.java

```diff
@@ -0,0 +1,366 @@
+        } else if (statusText.equals(TEMPORARY)) {
+            cancelCode = CANCEL_CODE_P;
+        } else if (statusText.equals(PERMANENT)) {
+            cancelCode = CANCEL_CODE_M;
+        } else if (statusText.equals(BANK_REQUEST)) {
+            cancelCode = CANCEL_CODE_B;
+        } else if (statusText.equals(CARDHOLDER_TERMINATED)) {
+            cancelCode = CANCEL_CODE_T;
+        } else if (statusText.equals(CUSTOMER_REQUEST)) {
+            cancelCode = CANCEL_CODE_R;
+        } else if (statusText.equals(COLLECTIONS)) {
+            cancelCode = CANCEL_CODE_C;
+        } else if (statusText.equals(NO_LONGER_NEEDED)) {
+            cancelCode = CANCEL_CODE_N;
+        } else if (statusText.equals(CARDHOLDER_LEFT_COMPANY)) {
+            cancelCode = CANCEL_CODE_A;
+        } else if (statusText.equals(CANCELED_WITHIN_FIVE_DAYS)) {
+            cancelCode = CANCEL_CODE_O;
+        } else if (statusText.equals(LOST_STOLEN)) {
+            cancelCode = CANCEL_CODE_L;
+        } else if (statusText.equals(LOST_STOLEN_FRAUD)) {
+            cancelCode = CANCEL_CODE_D;
+        } else {
+            cancelCode = CANCEL_CODE_X;
+        }
+        return cancelCode;
+        
+    }
+    
+    /**
+     * Determines the system ID from the employee ID and status test
+     */
+    private String determineSystemId (String employeeId, String statusText) {
+        //determine valid system id (net id) from employee id and status text
+        String systemId;
+        if (statusText.equals(ACTIVE) || statusText.equals(CARD_ACTIVATION) || statusText.equals(MONITORING) || statusText.equals(OPEN)) {
+            systemId = DEFAULT_SYS_ID;
+        }
+        else {
+            systemId = null;
+        }
+        
+        if (ObjectUtils.isNotNull(employeeId)) {
+            cardholderUser = getCardholderUser(employeeId.substring(1));
+            if (ObjectUtils.isNotNull(cardholderUser)) {
+                //employee found
+                systemId = cardholderUser.getPrincipalId();
+            }
+        }
+        return systemId;
+        
+    }
+      
+    /**
+     * Determines the expiration date
+     */
+    private Date determineExpireDate (String expireDate) {
+        //determine valid expiration date from string of mmdd
+        int expireMonth = Integer.parseInt(expireDate.substring(0, 2));
+        int expireDay = 31;
+        int expireYear = Integer.parseInt(expireDate.substring(2));
+        expireYear = expireYear + 2000;
+        if (expireMonth == 2) {
+            if (expireYear % 4 == 0)
+                expireDay = 29;
+            else
+                expireDay = 28;
+        }
+        if (expireMonth == 4 || expireMonth == 6 || expireMonth == 9 || expireMonth == 11) {
+            expireDay = 30;
+        }
+        
+        Date expDate = Date.valueOf(String.valueOf(expireYear) + "-" + expireDate.substring(0, 2) + "-" + String.valueOf(expireDay));
+        return expDate;
```

Comment:
Please add a LOG.debug statement at the end of the method to log the resulting expirationDate.